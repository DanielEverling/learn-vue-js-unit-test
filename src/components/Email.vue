<template>
  <div>
    <input type="text" name="email" v-model="email" @change="isEmail" />
    <span class="error" v-if="!validEmail">Informe um e-mail v√°lido</span>
  </div>
</template>

<script>
export default {
  data() {
    return {
      email: "",
      validEmail: true
    };
  },
  methods: {
    isEmail(email) {
      const expression = /^[a-z0-9.]+@[a-z0-9]+\.[a-z]+(\.[a-z]+)?$/i;
      return expression.test(email);
    },
  },
  watch: {
    email: function (value) {
      this.validEmail = this.isEmail(value)
    }
  },
};
</script>