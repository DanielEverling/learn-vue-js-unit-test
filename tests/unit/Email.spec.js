import { shallowMount } from '@vue/test-utils'
import Email from '@/components/Email'

describe('UserView', () => {

    const build = () => {
        const wrapper = shallowMount(Email)

        return {
            wrapper
        }
    }

    it('renders the component', () => {
        // arrange
        const { wrapper } = build()
        const inputEmail = wrapper.find('input[name="email"]') 
        const spanError = wrapper.find('span.error')

        // assert
        expect(wrapper.html()).toMatchSnapshot()      
        expect(inputEmail.exists()).toBe(true)
        expect(spanError.exists()).toBe(false)
    })

    it('should validate not valid email and render span with error message', () => {
        // arrange
        const { wrapper } = build()
        const inputEmail = wrapper.find('input[name="email"]') 
        inputEmail.setValue('invalid.email')

        const spanError = wrapper.find('span.error')
        
        // assert
        expect(wrapper.html()).toMatchSnapshot()
        expect(inputEmail.exists()).toBe(true)
        expect(spanError.exists()).toBe(true)
    })

    it('should validate valid email and not render span with error message', () => {
        // arrange
        const { wrapper } = build()
        const inputEmail = wrapper.find('input[name="email"]') 
        inputEmail.setValue('valid.email@test.com')

        const spanError = wrapper.find('span.error')
        
        // assert
        expect(wrapper.html()).toMatchSnapshot()
        expect(inputEmail.exists()).toBe(true)
        expect(spanError.exists()).toBe(false)
    })

})