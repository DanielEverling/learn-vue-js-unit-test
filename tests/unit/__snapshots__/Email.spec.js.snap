// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div><input type="text" name="email">
  <!---->
</div>
`;

exports[`UserView should validate not valid email and render span with error message 1`] = `<div><input type="text" name="email"> <span class="error">Informe um e-mail v√°lido</span></div>`;

exports[`UserView should validate valid email and not render span with error message 1`] = `
<div><input type="text" name="email">
  <!---->
</div>
`;
